Frameworks/Foundation/NSUndoManager.mm

```diff
@@ -38,11 +91,19 @@ - (id)prepareWithInvocationTarget:(id)target {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (void)undo {
-    UNIMPLEMENTED();
+    if (![self canUndo]) {
+        return;
+    }
+    _undoing = YES;
+
```

Comment:
No, NSUndoManager is not thread safe. 

---
In reply to: [75724843](https://github.com/Microsoft/WinObjC/pull/900#discussion_r75724843) [](ancestors = 75724843)